### ISSUE TITLE
Wrong password extracting zip file inside a zip file

### ISSUE THREAD

**Issue by Windolian** [srikanth-lingala/zip4j#87]
Hi.
I'm trying to extract a zip file (1234.zip - Password:1234) inside another zip file (123.zip, Password:1234), without extracting the first.
On myInputStreamFirmwarePackage.getNextEntry() 
and after like 6 files done correctly it crashes with error: "Wrong password...".

Here's the zip file
[123.zip](https://github.com/srikanth-lingala/zip4j/files/3724706/123.zip)

This is my code:

```
ZipFile myZipFile = new ZipFile("123.zip");
myZipFile.setPassword(caPasswordZip);

try {
    
			List<FileHeader> myFileHeaderList = myZipFile.getFileHeaders();

			for (int i = 0; i < myFileHeaderList.size(); i++) {

				FileHeader myFileHeader = (FileHeader) myFileHeaderList.get(i);
				if (myFileHeader != null) {

					if (myFileHeader.getFileName().equals("1234.zip")) {

						if (myZipFile.isEncrypted() == true) {

							myZipFile.setPassword("123".toCharArray());

							ZipInputStream myInputStream = new ZipInputStream(myZipFile
									.getInputStream(myFileHeader), "1234".toCharArray());

							LocalFileHeader localFileHeader;
							
							while ((localFileHeader = myInputStream.getNextEntry()) != null) { // After some files it crashes here with Wrong password even thought the password is all the same
                                while ((readLen = myInputStream.read(readBuffer)) != -1) {
								    
								  }
								System.out.println(localFileHeader.getFileName());
							}
							myInputStream.close();

						} else {

							Log.error("This is an unprotected file. Canceling file.");
							return false;
						}

						return true;
					}
				} else {

					Log.error("Fileheader is null. Shouldn't be here");
				}
			}
		} catch (ZipException e) {
		      e.printStackTrace();
			if (e.getType().equals(ZipException.Type.WRONG_PASSWORD)) {

				Log.error("Wrong password.");
				Log.error(e.getLocalizedMessage(), e);

			} else {

				Log.error(e.getLocalizedMessage(), e);
			}
		} catch (Exception e) {

			Log.error(e.getLocalizedMessage(), e);
		}

		return false;
```

Error is:
```
net.lingala.zip4j.exception.ZipException: Wrong Password
	at net.lingala.zip4j.crypto.AESDecrypter.init(AESDecrypter.java:76)
	at net.lingala.zip4j.crypto.AESDecrypter.<init>(AESDecrypter.java:50)
	at net.lingala.zip4j.io.inputstream.AesCipherInputStream.initializeDecrypter(AesCipherInputStream.java:36)
	at net.lingala.zip4j.io.inputstream.AesCipherInputStream.initializeDecrypter(AesCipherInputStream.java:17)
	at net.lingala.zip4j.io.inputstream.CipherInputStream.<init>(CipherInputStream.java:21)
	at net.lingala.zip4j.io.inputstream.AesCipherInputStream.<init>(AesCipherInputStream.java:31)
```

Those files are just a example. I'm doing it with bigger files and the error a bit diferent but the error type is still  ZipException.Type.WRONG_PASSWORD:
```
java.util.zip.DataFormatException: invalid stored block lengths
java.io.IOException: java.util.zip.DataFormatException: invalid stored block lengths
	at net.lingala.zip4j.io.inputstream.InflaterInputStream.read(InflaterInputStream.java:52)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.read(ZipInputStream.java:155)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.read(ZipInputStream.java:127)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.readUntilEndOfEntry(ZipInputStream.java:325)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:84)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:79)

```

**Comment by srikanth-lingala** [comment]
Before I try your example, I want to clarify if there is a typo in your code or in the issue mentioned here. In your issue you say that the password is `1234`, but in the code it is set as `myZipFile.setPassword("123".toCharArray());`

**Comment by Windolian** [comment]
Yes, my bad sorry.
But it doesn't change anything 
This line myZipFile.setPassword(caPasswordZip); is doing the same as this 
myZipFile.setPassword("123".toCharArray());

I set the 1234 password here

`ZipInputStream myInputStream = new ZipInputStream(myZipFile.getInputStream(myFileHeader), "1234".toCharArray());`

**Comment by srikanth-lingala** [comment]
I was able to reproduce your issue. However, I have created a new zip file (from some other files) on my machine with similar scenario (zip inside an encrypted zip), and I was able to extract it fine with your code. Seems like this issue has something to do with this zip file or the way it was generated.

**Comment by srikanth-lingala** [comment]
I will investigate and update this thread.

**Comment by srikanth-lingala** [comment]
I made a fix and released a snapshot version `2.2.3.5-SNAPSHOT`. Please try with this version and let me know if that fixes it. If yes, I will include this fix in the next final release. Please try not just with the zip file you attached (I tried and it works fine with the fix), but also with other combinations if possible. Thanks.

If you are unable to get the snapshot from maven, use the below in your pom in `distributionManagement`tag:

```
<snapshotRepository>
            <id>ossrh</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        </snapshotRepository>
```

**Comment by Windolian** [comment]
> 
> 
> I made a fix and released a snapshot version `2.2.3.5-SNAPSHOT`. Please try with this version and let me know if that fixes it. If yes, I will include this fix in the next final release. Please try not just with the zip file you attached, but also with other combinations if possible. Thanks.
> 
> If you are unable to get the snapshot from maven, use the below in your pom in `distributionManagement`tag:
> 
> ```
> <snapshotRepository>
>     <id>ossrh</id>
>     <url>https://oss.sonatype.org/content/repositories/snapshots</url>
> </snapshotRepository>
> ```

Can you give me the .jar file?

**Comment by srikanth-lingala** [comment]
You can download it from [here][1]. Sometimes the page loads a while to load.

[1]: https://oss.sonatype.org/content/repositories/snapshots/net/lingala/zip4j/zip4j/2.2.3.5-SNAPSHOT/

**Comment by srikanth-lingala** [comment]
Fixed in v2.2.3

**Comment by Windolian** [comment]
Sorry, was on vacations.
I now have this error when unzipping a zip file inside another zip file (both with password encrypted with AES-256).
They work if instead of AES-256 I use zipcrypto.
I zipped them with 7-zip.
```

java.lang.RuntimeException: AesExtraDataRecord not present in local header for aes encrypted data
	at net.lingala.zip4j.util.Zip4jUtil.getCompressionMethod(Zip4jUtil.java:92)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getCompressedSize(ZipInputStream.java:290)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.initializeEntryInputStream(ZipInputStream.java:199)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:106)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:79)
```
